### ISSUE TITLE
Unable to link different Cognito User Pools to different environments within same app

### ISSUE THREAD

**Issue by aizqdots** [aws-amplify/amplify-cli#10108]
### Before opening, please confirm:

- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-adminui/issues?q=is%3Aissue+).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-adminui/blob/main/CONTRIBUTING.md#bug-reports).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

### App Id

d1cyihgxi1td5r

### Region

eu-west-1

### Environment name

prd

### Amplify CLI Version

7.6.26

### If applicable, what version of Node.js are you using?

17.3.1

### What operating system are you using?

Mac

### Browser type?

Firefox & Chrome

### Describe the bug

Amplify is overriding/misconfiguring Cognito Authentication setup between the two environments I have in my app.

I have an Amplify application, with two environments and two Cognito User Pools (with their respective Identity Pools) created outside Amplify.

When I created the application and the first environment (dev), I imported using the Amplify CLI the first Cognito User Pool. Works as expected.

Then I created the second environment (prd) and I imported the second Cognito User Pool with Amplify CLI, but after deployment finishes, I see both environments configured only with the first User Pool.
 
<img width="644" alt="Captura de pantalla 2022-03-31 a las 14 23 54" src="https://user-images.githubusercontent.com/102737174/161054757-d6888272-aaf8-4f1d-9739-b6a433914261.png">

<img width="1037" alt="Captura de pantalla 2022-03-31 a las 14 15 15" src="https://user-images.githubusercontent.com/102737174/161053135-c02446a1-f10d-4243-8057-ef5d03ce867e.png">

And then the Admin UI for prd environment shows the configuration of the first User Pool:

![Captura de pantalla 2022-03-31 a las 14 30 49](https://user-images.githubusercontent.com/102737174/161055478-d0be2462-61fa-43e6-85cd-7a524ec01635.png)

I cannot import the second User Pool using Amplify Admin UI because it throws an error regarding Lambda environment variables not being properly set (this error only shows in Admin UI, not in CLI).

![Captura de pantalla 2022-03-31 a las 14 21 11](https://user-images.githubusercontent.com/102737174/161053522-8481f3b8-15ba-42e7-9c0f-04ec8e53ce60.png)

*Adding team-provider-info.json*
```
{
  "dev": {
    "awscloudformation": {
      "AuthRoleName": "amplify-xxx-authRole",
      "UnauthRoleArn": "arn:aws:iam::xxx:role/amplify-xxx-unauthRole",
      "AuthRoleArn": "arn:aws:iam::xxx:role/amplify-xxx-authRole",
      "Region": "eu-west-1",
      "DeploymentBucketName": "amplify-xxx-deployment",
      "UnauthRoleName": "amplify-xxx-dev-unauthRole",
      "StackName": "amplify-xxx-staging-xxx",
      "StackId": "arn:aws:cloudformation:eu-west-1:xxx:stack/amplify-xxx-staging-xxx/xxx",
      "AmplifyAppId": "---xi1td5r",
      "APIGatewayAuthURL": "https://s3.amazonaws.com/xxx"
    },
    "categories": {
      "auth": {
        "authName1": {
          "userPoolId": "eu-west-1_xxx2Xsv",
          "userPoolName": "pool1",
          "webClientId": "xxx",
          "nativeClientId": "xxx",
          "identityPoolId": "eu-west-1:xxxb4e9d",
          "identityPoolName": "idpXXX-Pre",
          ...
        }
      },
      "function": {
        "func1": {
          "deploymentBucketName": "amplify-xxx-staging-xxx-deployment",
          "s3Key": "amplify-builds/xxx-xxx-build.zip"
        },
        "func2": {
          "deploymentBucketName": "amplify-xxx-staging-xxx-deployment",
          "s3Key": "amplify-builds/xxx-xxx-build.zip",
          "envVar1": "xxx",
          "envVar2": "xxx"
        }
      }
    }
  },
  "prd": {
    "awscloudformation": {
      "AuthRoleName": "amplify-xxx-prd-xxx-authRole",
      "UnauthRoleArn": "arn:aws:iam::xxx:role/amplify-xxx-prd-xxx-unauthRole",
      "AuthRoleArn": "arn:aws:iam::xxx:role/amplify-xxx-prd-xxx-authRole",
      "Region": "eu-west-1",
      "DeploymentBucketName": "amplify-xxx-prd-xxx-deployment",
      "UnauthRoleName": "amplify-xxx-prd-xxx-unauthRole",
      "StackName": "amplify-xxx-prd-xxx",
      "StackId": "arn:aws:cloudformation:eu-west-1:xxx:stack/amplify-xxx-prd-234344/xxx-12312",
      "AmplifyAppId": "----1td5r",
      "APIGatewayAuthURL": "https://s3.amazonaws.com/amplify-xxx-prd-xxx-deployment/..."
    },
    "categories": {
      "function": {
        "func1": {
          "deploymentBucketName": "amplify-xxx-prd-xxx-deployment",
          "s3Key": "amplify-builds/xxx-xxx-build.zip"
        },
        "func2": {
          "deploymentBucketName": "amplify-xxx-prd-xxx-deployment",
          "s3Key": "amplify-builds/xxx-xxx-build.zip",
          "envVar1": "xxx",
          "envVar2": "xxx"
        }
      },
      "auth": {
        "authName2": {
          "userPoolId": "eu-west-1_xxxxPWZq",
          "userPoolName": "xxx-Pro-UP",
          "webClientId": "xxxc0vp",
          "nativeClientId": "xxxxc0vp",
          "identityPoolId": "eu-west-1:xxx57991c",
          "identityPoolName": "xxx-Pro",
          ...
        }
      }
    }
  }
}
```

### Expected behavior

When importing the second User Pool Auth(xxx-PRO-UP), it should be linked to prd environment instead of PRE environment.

### Reproduction steps

 1. Have an Amplify App and two Cognito User Pools created manually
 2. Create the first environment, and link it to the first Cognito User Pool
 amplify init
 amplify import auth (and follow steps)
 git add . && git push
 3. Create the second environment and import second User Pool
 git checkout -b prd 
 amplify env add prd
 amplify checkout prd
 amplify import auth (and follow steps for 2nd UP)
 git add .
 git push

### Additional information

I tried to unlink and relink the User Pool for PRD environment several times without success

**Comment by ykethan** [comment]
Hello @aizqdots, Thank you for reaching out. I was curious on the `amplify env add prd`, on executing the command the CLI will prompt the following.

`
? The resource: 'congtioxxx' (Cognito User Pool: 'us-east-1_xxxx') already imported to 'dev' environment, do you want to import it to the new environment? (Y/n) » Yes
If you choose No, then an import walkthrough will run to import a different resource into the new environment.
`

If selected `yes` the environment will import the existing Cognito resource. If selected `no`, the CLI will prompt the Cognito user pools to select from. Could you please let me if a new user pool was selected in this step or the user pool from the previous environment was selected. 

As when performing the `amplify auth` the CLI will throw the following warning.

`Auth has already been imported to this project and cannot be modified from the CLI. To modify, run "amplify remove auth" to unlink the imported 
auth resource. Then run "amplify import auth".`

**Comment by aizqdots** [comment]
Hello, sorry for taking so long to answer.
I selected "no", because user pools are different for each environment.

This issue was solved when I set the missing Lambda environment variables with the Amplify CLI  in the second environment (prd). It seems it is related with this [other issue](https://github.com/aws-amplify/amplify-cli/issues/8111), but in my case the console wasn't asking for the missing variables.
